documentcloud/documents/processing/ocr/main.py

```diff
@@ -71,34 +83,100 @@
 LARGE_IMAGE_SUFFIX = "-large"
 TXT_EXTENSION = ".txt"
 
+TESS_PDF_PREFIX = "ocr"
+PDF_FONT_FILE = "pdf.ttf"  # Tesseract invisible font
+
 
 def write_text_file(text_path, text, access):
     """Helper method to write a text file."""
     storage.simple_upload(text_path, text.encode("utf8"), access=access)
 
 
-def ocr_page(page_path, ocr_code="eng"):
+def local_folder_size(folder_path):
+    return (
+        sum(file.stat().st_size for file in Path(folder_path).rglob("*")) / 1024 / 1024
+    )
+
+
+def download_tmp_file(relative_path):
+    """Downloads the requested data file to a tmp directory."""
+    local_file_path = os.path.join(TMP_DIRECTORY, relative_path)
+    if os.path.exists(local_file_path):
+        # OCR language pack already downloaded
+        return
+
+    # Check if tmp directory is too big
+    if local_folder_size(TMP_DIRECTORY) > TMP_SIZE_LIMIT:
+        # If so, just delete all OCR data (shouldn't happen too often)
+        files = Path(TMP_DIRECTORY).rglob("*")
+        for file in files:
+            os.remove(file)
```

Comment:
If we are potentially clearing out `TMP_DIRECTORY` probably best to put it in its own subdirectory in `/tmp` if its easy enough

Reply:
Maybe log something if this happens, just so we can check up on it